packages/ui/src/components/Dialog/Dialog.css.ts

```diff
@@ -0,0 +1,51 @@
+import { keyframes, style, styleVariants } from '@vanilla-extract/css'
+import { animationAllowed, theme } from '../../theme'
+
+const overlayShow = keyframes({
+  '0%': { opacity: 0 },
+  '100%': { opacity: 1 },
+})
+
+export const overlay = styleVariants({
+  base: {
+    backgroundColor: 'rgba(0, 0, 0, 0.15)',
+    position: 'fixed',
+    inset: 0,
+    '@media': {
+      [animationAllowed]: {
+        animation: `${overlayShow} 500ms cubic-bezier(0.16, 1, 0.3, 1) forwards`,
+      },
+    },
+  },
+  frosted: {
+    backgroundColor: theme.colors.backgroundFrostedGlass,
+    backdropFilter: 'blur(64px)',
+  },
+})
```

Comment:
[Discussion]
What do you think about the other notation where we add `@media` rules at the same level of all styles? Like we do for emotion styled components today. Should we set a common preference?

```
// theme
export const animationAllowed = '@media (prefers-reduced-motion: no-preference)'
```

```suggestion
export const overlay = styleVariants({
  base: {
    backgroundColor: 'rgba(0, 0, 0, 0.15)',
    position: 'fixed',
    inset: 0,
    [animationAllowed]: {
      animation: `${overlayShow} 500ms cubic-bezier(0.16, 1, 0.3, 1) forwards`,
    },
  },
  frosted: {
    backgroundColor: theme.colors.backgroundFrostedGlass,
    backdropFilter: 'blur(64px)',
  },
})
```

Reply:
Started with that, decided against it for 2 reasons
- type safety.  Not compatible with vanilla types unless you specifically downcast media rules to generic `string` which makes code more awkward
- compilation results are different.  Both variants work in Chrome, but I would not be 100% sure about all browsers